File: articles/resilience-thresholds-and-regime-shifts/_archive/pre-no-pandas-professional-fix-20260531-195617/python/run_all_resilience_workflows.py
#!/usr/bin/env python3
"""Run all professional Python workflows for the resilience article."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ROOT = Path(__file__).resolve().parents[1]
SCRIPTS = [
    "resilience_threshold_regime_model.py",
    "early_warning_indicator_analysis.py",
    "hysteresis_recovery_diagnostics.py",
    "adaptive_capacity_scenarios.py",
    "resilience_sensitivity_analysis.py",
    "distributional_vulnerability_model.py",
    "export_resilience_outputs.py",
    "validation_checks.py",
]


def main() -> None:
    for script in SCRIPTS:
        print(f"\n=== Running {script} ===")
        subprocess.run([sys.executable, str(ROOT / "python" / script)], check=True)
    print("\nAll Python resilience workflows completed successfully.")


if __name__ == "__main__":
    main()

File: articles/resilience-thresholds-and-regime-shifts/_archive/pre-no-pandas-professional-fix-20260531-195617/r/run_all_resilience_workflows.R
# Run all professional R workflows for the resilience article.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)
if (length(file_arg) > 0) {
  script_dir <- dirname(sub("^--file=", "", file_arg[1]))
  root <- normalizePath(file.path(script_dir, ".."), mustWork = FALSE)
} else {
  root <- getwd()
}
setwd(root)

source(file.path("r", "resilience_threshold_visualization.R"))
source(file.path("r", "early_warning_diagnostics.R"))

cat("All R resilience workflows completed successfully.\n")
